pkg/clients/networking.go

```diff
@@ -238,6 +239,40 @@ func (c networkClient) DeleteTrunk(id string) error {
 	return mc.ObserveRequestIgnoreNotFound(trunks.Delete(c.serviceClient, id).ExtractErr())
 }
 
+func (c networkClient) RemoveTrunkSubPorts(id string) error {
```

Comment:
This would be a significant change of pattern for `clients`. So far we've confined clients to being an extremely thin wrapper round a gophercloud call, essentially so we can mock them.

The disadvantage of this is that we can't mock it. Would you mind moving this logic into trunk.go and calling existing client methods instead?

I'm not completely opposed to going this way, btw. The 'interfaces' we have in `cloud/services/*` are ill-defined at best, chaotic at worst.  However, if we're going to clean it up I'd prefer to do it deliberately.